src/Validation/ValidatorAwareTrait.php

```diff
@@ -95,8 +94,8 @@ public function validator($name = null, Validator $validator = null)
         }
 
         if ($validator === null) {
-            $validator = new Validator();
-            $validator = $this->{'validation' . ucfirst($name)}($validator);
+            $className = isset($this->_validatorClass) ? $this->_validatorClass : '\Cake\Validation\Validator';
```

Comment:
Can't the trait define this property? Its a bit odd that the property is in the implementing class.